draft-gpew-priv-ppm.md

```diff
@@ -672,247 +677,377 @@ independently.
-  reports included in the aggregation, then combining the hash values with a
-  bitwise-XOR operation.
-* `helper_state` is the helper's state, which is carried across requests from
-  the leader.
+For each aggregator endpoint `[aggregator]` in `AggregateContinueReq.task_id`'s
+parameters except its own, the leader sends a POST request to
+`[aggregator]/aggregate` with AggregateContinueReq as the payload and the media
+type set to "message/ppm-aggregate-continue-req".
 
-To handle the leader's request, the helper first ensures that the request meets
-the requirements for batch parameters following the procedure in
-{{batch-parameter-validation}}. If the batch parameters are invalid, then it
-MUST abort with error "batchInvalid".
+#### Helper Continuation
 
-To respond to an AggregateShareReq message, the helper first looks up the PPM
-parameters associated with task `task_id`. Then, using the procedure in
-{{batch-parameter-validation}}, it ensures that the request meets the
-requirements of the batch parameters. It also computes a checksum based on its
-view of the output shares included in the batch window, and checks that the
-`report_count` and `checksum` included in the request match its computed
-values. If so, it aggregates all valid output shares that fall in the batch
-interval into an aggregate share. The response contains an opaque,
-VDAF-specific message:
+The helper continues with preparation for a report share by combining the leader's input
+message in `AggregateContinueReq` and its current preparation state (`prep_state`). This
+step yields one of three outputs:
+
+1. An error, in which case the input report share is marked as invalid.
+1. An output share, in which case the helper stores the output share for future collection
+   as described in {{collect-flow}}.
+1. An updated VDAF state and preparation message, denoted `(prep_state, prep_msg)`.
+
+To carry out this step, for each PrepareStep in AggregateContinueReq.prepare_shares received
+from the leader, the helper performs the following check to determine if the report share
+should continue being prepared.
+
+* If failed, then mark the report as failed and reply with a failed PrepareStep
+  to the leader.
+* If finished, then mark the report as finished and reply with a finished PrepareStep
+  to the leader. The helper then stores the output share and awaits for collection;
+  see {{collect-flow}}.
+
+Otherwise, preparation continues. In this case, the helper computes its updated state
+and output message as follows:
 
 ~~~
-struct {
-  opaque message<0..2^16-1>; // VDAF message
-} AggregateShare;
+out = VDAF.prep_next(prep_state, inbound)
 ~~~
 
-Next, the helper encrypts the aggregate share `agg_share` under the collector's
-public key as follows:
+where `inbound` is the previous VDAF preapre message sent by the leader and `prep_state` is
+its current preparation state. If this operation fails, then the helper fails
```

Comment:
```suggestion
the helper's current preparation state. If this operation fails, then the helper fails
```
"its" is ambiguous since the sentence earlier refers to the leader.